ga4gh/backend.py

```diff
@@ -17,6 +17,81 @@
 import ga4gh.datamodel.variants as variants
 
 
+def _getReadAlignmentStart(readAlignment):
+    return readAlignment.alignment.position.position
+
+
+def _getReadAlignmentEnd(readAlignment):
+    return _getReadAlignmentStart(readAlignment) + \
+        len(readAlignment.alignedSequence)
+
+
+def _getVariantStart(variant):
+    return variant.start
+
+
+def _getVariantEnd(variant):
+    return variant.end
+
+
+class IntervalIterator(object):
+    """
+    Implements generator logic for types which accept a start/end
+    range to search for the object
+    """
+    def __init__(self, request, iterator, getStart, getEnd):
+        self._request = request
+        self._iterator = iterator
+        self._getStart = getStart
+        self._getEnd = getEnd
+        self._badPageTokenExceptionMessage = (
+            "Inconsistent page token provided")
+
+    def iterate(self, startPosition, equalPositionsToSkip):
```

Comment:
`__next__` is only for python3.  python 2 requires `next`.  We should probably avoid implementing this method if we're worried about compatibility (even in the distant future), in any event.